### PR TITLE
Action list reorder

### PR DESCRIPTION
**Pull request by tashfia98**
### Summary 
- Added the drag and drop functionality to the Action Sequence List.
- Can reorder the buttons on Action Sequence List component. 

### Details
- Added react-beautiful-dnd react library using yarn to be able to make the drag and drop feature. 
- The list in the Action Sequence List can be reordered via drag and drop.
- The elements can be dragged across other components but only locks the position when dropped in Action Sequence List component.
- Jira issue ([EDEX-338](https://atomata.atlassian.net/browse/EDEX-338))

### How to test?
1. Run development side and click "EVIL-CYLINDER" button on the start up page. 
2. Once it loads up, click the "Toggle Overlay" button to see the react components. 
3. Click "WOBBLE-SPHERE" button from the Apparatus List component.
4. Click the "+" button beside the actions in the Action List component to add actions in the  Action Sequence List component.
5. Once the buttons shows up in  Action Sequence List component, drag & drop to reorder and test the feature. 
6. Test drag & drop again by removing the buttons from Action Sequence List component and re adding as well.

### Checks 
- [x] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

**Comment by Ysabz** [comment]
@tashfia98 , when I reorder the elements, and then delete, the wrong item will be deleted

File: src/components/apparatusLists/ActionSequenceBox.tsx [hunk 1]
```diff
@@ -41,26 +42,71 @@ const ListBoxScroller = styled.div`
...
 const ActionSequenceBox = ({ actionList, removeAction }) => {
+
+  //Setting the array from the parameter with hook.
+  const [list,setList] = useState(actionList);
+  if (list.length != actionList.length) {
+    setList(actionList);
+  }
+
+  //Settig the postion of the item dragged and dropped.
+  function handleOnDragEnd (result) {
```
Comment (Ysabz) [inline src/components/apparatusLists/ActionSequenceBox.tsx:53]: it's better to perform changes to the action list from overlay which is the parent component. for example, to remove an action from list, a function in overlay is called. doing this also removes the need to define your state variables and using props for setting the variables

**Review by Ysabz — CHANGES_REQUESTED** [review]

File: src/components/apparatusLists/ActionSequenceBox.tsx (hunk 1 above)
Comment (tashfia98) [reply src/components/apparatusLists/ActionSequenceBox.tsx:46]: Thank you for your review! I made changes based on your suggestion. Can you please test it out and let me know if your still having the issue?

**Comment by shushengfuqin** [comment]
Tested.
Function wise. LGTM

File: src/components/Overlay/Overlay.tsx [hunk 2]
```diff
@@ -101,6 +101,21 @@ function Overlay({ json }: { json: any }): JSX.Element {
...
+   //Setting the postion of the item dragged and dropped.
+   function handleOnDragEnd (result) {
+
+    //dropped outside the list
+    if (!result.destination) {
+      return;
+    }
+
+    //console.log(result);
+    const items = actionList;
```
Comment (Ysabz) [inline src/components/Overlay/Overlay.tsx:113]: you can directly use actionlist

**Review by Ysabz — APPROVED** [review]

File: src/components/apparatusLists/ActionSequenceBox.tsx (hunk 1 above)
Comment (Ysabz) [reply src/components/apparatusLists/ActionSequenceBox.tsx]: Tested. It works

Comment (tashfia98) [reply src/components/apparatusLists/ActionSequenceBox.tsx]: Thanks!

**Pull request merged**